scripts/git/pre-commit

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+
+CHANGES=$(git diff --name-only origin/main)
+
+
+if [ -z "$CHANGES" ]
+then
+  echo "No changes"
+  exit 1
+fi
+
+CONTRACT_NAME=lst-dex-adapter-osmosis
+if [[ $CHANGES == *contracts/${CONTRACT_NAME}* ]]
+then
+  echo "Changes in ${CONTRACT_NAME} contract"
+
+  cd ${REPO_ROOT}/smart-contracts/contracts/${CONTRACT_NAME}
+  cargo run --bin schema --features schema
+  NEW_CHANGES=$(git diff --name-only origin/main)
+  if [[ $CHANGES == *contracts/${CONTRACT_NAME}/schema* ]]
+  then
+    git add schema
+  fi
+
+  cargo fmt --all
+  cd ${REPO_ROOT}
+  git add $CHANGES
+fi
+
+CONTRACT_NAME=dex-router-osmosis
+if [[ $CHANGES == *contracts/${CONTRACT_NAME}* ]]
+then
+  echo "Changes in ${CONTRACT_NAME} contract"
```

Comment:
This looks kinda redundant respect ht eprevious one, also, is veruy specific. Can we somehow generalize the --feature flag and so iterate automatically all the `contracts/**` sub-folders and generate schema?

If iterating is too much, lets just hardcode contract names.

Reply:
Introduced a function (now properly using `cargo schema` and `.cargo/config.toml`) to update schemas and fix formatting. 
For now I did hard-code the contracts that check. I also like the idea of automatically going over all contracts. Before I enable that, I would prefer to clean up the commented-out contracts, i.e. either remove them or enable them again.